spec/unit/changelog_source_spec.rb

```diff
@@ -0,0 +1,178 @@
+      let(:invalid_version) {
+        "* Add experimental `--exclude` option to exclude elements from inspection."
+      }
+
+      it "returns false" do
+        expect(ChangelogSource.parse_element(invalid_version)).to be_falsy
+      end
+    end
+  end
+
+  describe "#elements" do
+    let(:rpm_changes_expected_output) {
+      {
+        "1.6.0" => {
+          version: "1.6.0",
+          date: "Wed Apr 01 15:43:17 CEST 2015",
+          email: "johndoe@suse.de",
+          items: [
+            "update to version 1.6.0"
+          ]
+        },
+        "1.5.0" => {
+          version: "1.5.0",
+          date: "Fri Mar 13 13:03:47 CET 2015",
+          email: "johndoe@suse.de",
+          items: [
+            "update to version 1.5.0",
+            "tracking bug (bnc#914712)"
+          ]
+        }
+      }
+    }
+
+    let(:news_expected_output) {
+      { "1.6.0" =>
+        {
+          version: "1.6.0",
+          date: "Wed Apr 01 15:43:17 CEST 2015",
+          email: "johndoe@suse.de",
+          items: [
+            "Add experimental `--exclude` option to exclude elements from inspection.\n" \
+            "  Run `machinery config experimental-features on` to enable the option",
+            "Support global `--exclude` option for `show` command.\n" \
+            "  This feature is available when the experimental-features option is on.",
+            "Add `--verbose` option to `inspect` command to display the filters which\n" \
+            "  were used during inspection",
+            "Add `--verbose` option to `show` command to display the filters which were\n" \
+            "  applied before showing the system description",
+            "Show progress on upgrading the system description format",
+            "Improve help for users when upgrading system descriptions",
+            "Fix piping output of Machinery to less (gh#SUSE/machinery#521)",
+            "Better help for users when upgrading format of system descriptions",
+            "Add option to show command to display filters used during inspection",
+            "Support negated filter expressions in --exclude option",
+            "Show filters used during inspection in HTML view"
+          ]
+        },
+        "1.5.0" =>
+        {
+          version: "1.5.0",
+          date: "Fri Mar 13 13:03:47 CET 2015",
+          email: "johndoe@suse.de",
+          items: [
+            "Fix issue with changes of managed files on RHEL (gh#SUSE/machinery#636)",
+            "Add `--short` option to list command for showing a short list of all descriptions",
+            "Improve error output for system descriptions which have outdated formats",
+            "Add `--skip-files` option to `inspect` command for unmanaged files"
+          ]
+        }
+      }
+    }
+
+    before do
+      subject.parse
+    end
+
+    context "when parsing the rpm changes" do
+      let(:file) { File.join(File.dirname(__FILE__), "..", "data", "changelog", "RPM_CHANGES") }
```

Comment:
It would be nice to have a helper method to find a file in the `spec/data/` directory. This is repeated at several places in this patch. Extracting it as a method would make the tests easier to read.